server/opts.go

```diff
@@ -373,20 +373,54 @@ type token interface {
 // unwrapValue can be used to get the token and value from an item
 // to be able to report the line number in case of an incorrect
 // configuration.
-func unwrapValue(v interface{}) (token, interface{}) {
+// also stores the token in lastToken for use in convertPanicToError
+func unwrapValue(v interface{}, lastToken *token) (token, interface{}) {
 	switch tk := v.(type) {
 	case token:
+		if lastToken != nil {
+			*lastToken = tk
+		}
 		return tk, tk.Value()
 	default:
 		return nil, v
 	}
 }
 
+// use in defer to recover from panic and turn it into an error associated with last token
+func convertPanicToErrorList(lastToken *token, errors *[]error) {
+	// only recover if an error can be stored
+	if errors == nil {
+		return
+	} else if err := recover(); err == nil {
+		return
+	} else if lastToken != nil && *lastToken != nil {
+		*errors = append(*errors, &configErr{*lastToken, fmt.Sprint(err)})
+	} else {
+		*errors = append(*errors, fmt.Errorf("Encountered panic without a token %v", err))
+	}
+}
+
+// use in defer to recover from panic and turn it into an error associated with last token
+func convertPanicToError(lastToken *token, e *error) {
+	// only recover if an error can be stored
+	if e == nil || *e != nil {
+		return
+	} else if err := recover(); err == nil {
+		return
+	} else if lastToken != nil && *lastToken != nil {
+		*e = &configErr{*lastToken, fmt.Sprint(err)}
+	} else {
+		*e = fmt.Errorf("%v", err)
+	}
+}
+
 // configureSystemAccount configures a system account
 // if present in the configuration.
-func configureSystemAccount(o *Options, m map[string]interface{}) error {
+func configureSystemAccount(o *Options, m map[string]interface{}) (err error) {
```

Comment:
Personal preference, but I would have called this one `retErr error` (and obviously pass that to convertPanicToError() down below), to indicate that this is the returned error and avoid possible shadowing in the rest of the function.

Reply:
changed